### PR TITLE
Made loading profile info more robust

### PR DESCRIPTION
**Pull request by rakhad**

### CODE REVIEW

File: Assets/Xbox Live/Scripts/UserProfile.cs [hunk 1]
```diff
@@ -206,31 +207,78 @@ public IEnumerator SignInAsync()
         // Throw any exceptions if needed.
         if (signInStatus == SignInStatus.Success)
         {
+            if (!subscribedToEventHandler)
```
Comment (jplafonta) [inline Assets/Xbox Live/Scripts/UserProfile.cs]: I'm confused why you are subscribing to this event differently depending on whether the user is already signed in or not.  It seems like we care either way.  Could you just subscribe to this event once in Start() and get rid of the subscribedToEventHandler variable?

Comment (rakhad) [reply Assets/Xbox Live/Scripts/UserProfile.cs]: Sure

File: Assets/Xbox Live/Scripts/UserProfile.cs [hunk 2]
```diff
@@ -206,31 +207,78 @@ public IEnumerator SignInAsync()
...
+        {
+            SocialManagerComponent.Instance.EventProcessed -= SocialManagerEventProcessed;
+            subscribedToEventHandler = false;
         }
     }
 
     private IEnumerator FinishLoadingProfileInfo()
     {
-        SocialManagerComponent.Instance.EventProcessed -= SocialManagerEventProcessed;
+        if (subscribedToEventHandler)
```
Comment (jplafonta) [inline Assets/Xbox Live/Scripts/UserProfile.cs]: Whats the rationale to unsubscribing from the event here?
and nulling out the user group below

File: Assets/Xbox Live/Scripts/UserProfile.cs [hunk 3]
```diff
@@ -206,31 +207,78 @@ public IEnumerator SignInAsync()
...
+            SocialManagerComponent.Instance.EventProcessed += SocialManagerEventProcessed;
+            subscribedToEventHandler = true;
+        }
+
         userGroup = XboxLive.Instance.SocialManager.CreateSocialUserGroupFromList(this.XboxLiveUser.User, new List<string> { this.XboxLiveUser.User.XboxUserId });
     }
 
     private void SocialManagerEventProcessed(object sender, SocialEvent socialEvent)
     {
-        if (socialEvent.EventType == SocialEventType.SocialUserGroupLoaded &&
```
Comment (jplafonta) [inline Assets/Xbox Live/Scripts/UserProfile.cs]: You might want to also check that these events are relevant to this user

**Pull request merged**